pkg/controllers/jivavolume_controller.go

```diff
@@ -873,6 +864,42 @@ func createReplicaStatefulSet(r *JivaVolumeReconciler, cr *openebsiov1alpha1.Jiv
 				if cr.Spec.Policy.Replica.NodeSelector != nil {
 					ptsBuilder = ptsBuilder.WithNodeSelector(cr.Spec.Policy.Replica.NodeSelector)
 				}
+				if cr.Spec.Policy.Replica.Affinity != nil {
+					if cr.Spec.Policy.Replica.Affinity.PodAntiAffinity != nil {
+						for _, term := range cr.Spec.Policy.Replica.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution {
+							selectorMap, _ = metav1.LabelSelectorAsMap(term.LabelSelector)
+						}
+						defaultLabels[replicaAntiAffinityKey] = selectorMap[replicaAntiAffinityKey]
+					}
+				}
+				ptsBuilder = ptsBuilder.WithLabels(defaultLabels)
+				affinity := &corev1.Affinity{
+					PodAntiAffinity: &corev1.PodAntiAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+							{
+								LabelSelector: &metav1.LabelSelector{
+									MatchLabels: defaultReplicaMatchLabels(cr.Spec.PV),
```

Comment:
```suggestion
									MatchLabels: defaultLabels,
```

Reply:
This is affnity term , which uses the `defaultReplicaMatchLabels` , same as earlier